packages/from_interface_derive/src/lib.rs

```diff
@@ -0,0 +1,66 @@
+extern crate proc_macro;
+
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, ItemEnum};
+const INTERFACE_POSTFIX: &str = "Interface";
+
+// FIXME: it doesn't support generics (we don't need it right now )
+#[proc_macro_derive(FromInterface, attributes(renamed))]
+pub fn from_derive(input: TokenStream) -> TokenStream {
+    let ast = parse_macro_input!(input as ItemEnum);
+    let name = &ast.ident;
+    let counter_name = {
+        let counter_name = name.to_string();
```

Comment:
Ok, very nice ! 
Could we rename the `counter_name` to something else, maybe `original_name`, because `counter` doesn't exist here I think